cpp/util/profiler.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2015-2015 the openage authors. See copying.md for legal info.
+
+#ifndef OPENAGE_UTIL_PROFILER_H_
+#define OPENAGE_UTIL_PROFILER_H_
+
+#include <map>
+#include <vector>
+#include <chrono>
+#include <list>
+
+constexpr int MAX_DURATION_HISTORY = 100;
+
+namespace openage {
+namespace util {
+
+struct CategorieTimeData {
+	std::chrono::high_resolution_clock::time_point start;
+	std::chrono::high_resolution_clock::duration duration;
+	std::list<std::chrono::high_resolution_clock::duration> history;
+};
+
+class Profiler {
+public:
+	enum class Categorie {
```

Comment:
enum class names are lowercased again, also the correct word is `category`, but you might want to name it `component` or `subsystem`. which brings me to the idea to have some general subsystem registry which we can then use for the profiler.